src/Model/Message/MessageBag.php

```diff
@@ -4,19 +4,26 @@
 
 namespace PhpLlm\LlmChain\Model\Message;
 
-/**
- * @template-extends \ArrayObject<int, MessageInterface>
- */
-final class MessageBag extends \ArrayObject implements \JsonSerializable
+final class MessageBag implements \Countable, \JsonSerializable
 {
+    /**
+     * @var MessageInterface[]
+     */
+    public array $messages = [];
```

Comment:
if we directly expose the array we cannot ensure any type safety here. everyone could (def including me) skip the `add(...)` method and just write on the array. no matter if `MessageInterface` or not.

too scared, would prefer to be more defensive here and have a `getMessages()` instead

Reply:
done